addons/mail/models/mail_mail.py

```diff
@@ -576,6 +577,92 @@ def _split_by_mail_configuration(self):
             for batch_ids in tools.split_every(batch_size, record_ids):
                 yield mail_server_id, alias_domain_id, smtp_from, batch_ids
 
+    def _split_by_delayed_batch(self, mail_server):
+        """To not flag personal email servers as spam, we throttle them at X emails / minutes."""
+        if not mail_server or not mail_server.owner_user_id:
+            return self
+
+        # Send the oldest mails in priority if the CRON is late
+        mails = self.browse()
+        notifs = self.env['mail.notification'].search([
+            ('notification_type', '=', 'email'),
+            ('mail_mail_id', 'in', self.ids),
+            ('notification_status', 'not in', ('sent', 'canceled'))
+        ])
+        for mail in self.sorted(lambda k: (k.create_date, k.id)):
+            if len(mail.recipient_ids) > 1:
```

Comment:
Shouldn't we accumulate recipient count and split only the last mail that is going to make the counter overflow ? and let other untouched. Would make it a bit less intrusive, and split only the "last possible mail". Because I don't like that kind of magic processing, would make it as less intrusive as possible ^^ .